### PR TITLE
fix(badge): Badge should display count when color and count are both exist

### LINKED ISSUE #37542: Badge 的 count 属性不能和 color 一起用, 不会展示 count
### Reproduction link

count 和 color 一起用就不会展示 count, 要改变背景就只能用 style={{ backgroundColor: '#52c41a' }}
https://codesandbox.io/s/duo-cai-hui-biao-antd-4-23-1-forked-tem2cw

### What is expected?

显示一个有 count 数量的颜色点

### What is actually happening?

只显示一个颜色点, 没有 count 文字

| Environment | Info |
| --- | --- |
| antd | 4.23.1 |
| React | 18.2.0 |
| System | macos |
| Browser | chrome |

### PR DESCRIPTION
**Pull request by kiner-tang**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [X] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link

- fix: #37542 

### 💡 Background and solution

image

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |    Badge should display count when color and count are both exist       |
| 🇨🇳 Chinese |     同时指定 color 和 count 属性时，Badge 组件支持显示 count      |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before review. ⚠️

- [ ] Doc is updated/provided or not needed
- [X] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [X] Changelog is provided or not needed

### CODE REVIEW

**Comment by afc163** [comment]
rebase to fix ci

File: components/badge/demo/colorful.md [hunk 1]
```diff
@@ -36,13 +36,35 @@ const colors = [
...
+                  marginBottom: 10,
+                }}
+              >
+                {color}
+              </div>
+            </Badge>
+          </div>
+        ))}
+      </Col>
+    </Row>
```
Comment (afc163) [inline components/badge/demo/colorful.md:67]: 单独放一个例子，加 debug: true ，不作为默认推荐的用法。

**Comment by afc163** [comment]
发到 feature 分支吧。

**Comment by kiner-tang** [comment]
> 发到 feature 分支吧。

重新提了一个到`feature`的`PR`:

- https://github.com/ant-design/ant-design/pull/37609

**Pull request closed without merging**